packages/contract/README.md

```diff
@@ -10,51 +10,51 @@ $ npm install @truffle/contract
+const MyContract = require("./path/to/contractArtifact.json"); //produced by truffle compile
+const contract = require("@truffle/contract");
 
-var MyContract = contract({
-  abi: ...,
-  address: ..., // optional
-  // many more
-})
-MyContract.setProvider(provider);
+const instance = contract(MyContract);
 ```
 
-You now have access to the following functions on `MyContract`, as well as many others:
+If you aren't using truffle artifacts, the input to the `contract` function is a JSON blob defined by [@truffle/contract-schema](https://github.com/trufflesuite/truffle/tree/master/packages/contract-schema). This JSON blob is structured in a way that can be passed to all Truffle-related projects.
+Minimally, it requires the contract ABI, but passing this alone may cause issues using certain features of truffle contract.
 
-* `at()`: Create an instance of `MyContract` that represents your contract at a specific address.
-* `deployed()`: Create an instance of `MyContract` that represents the default address managed by `MyContract`.
-* `new()`: Deploy a new version of this contract to the network, getting an instance of `MyContract` that represents the newly deployed instance.
+```javascript
+const instance = contract({
+  abi: [...], // minimum required
+  address: "0x...", // optional
+  // many more
+});
+instance.setProvider(provider);
+```
 
-Each instance is tied to a specific address on the Ethereum network, and each instance has a 1-to-1 mapping from Javascript functions to contract functions. For instance, if your Solidity contract had a function defined `someFunction(uint value) {}` (solidity), then you could execute that function on the network like so:
+You now have access to the following functions on `instance`, as well as many others:
 
-  ```javascript
-  var deployed;
-  MyContract.deployed().then(function(instance) {
-    deployed = instance;
-    return instance.someFunction(5);
-  }).then(function(result) {
-    // Do something with the result or continue with more transactions.
-  });
-  ```
+- `at()`: Create an instance of `MyContract` that represents your contract at a specific address.
+- `deployed()`: Create an instance of `MyContract` that represents the default address managed by `MyContract`.
+- `new()`: Deploy a new version of this contract to the network, getting an instance of `MyContract` that represents the newly deployed instance.
 
-or equivalently in ES6 <sup>(node.js 8 or newer)</sup>:  
+Each instance is tied to a specific address on the Ethereum network, and each instance has a 1-to-1 mapping from Javascript functions to contract functions. For instance, if your Solidity contract had a function defined `someFunction(uint value) {}` (solidity), then you could execute that function on the network like so:
 
-  ```javascript
-  const instance = await MyContract.deployed();
-  const result = await instance.someFunction(5);  
-  ```  
+```javascript
+const deployed;
+const instance = await MyContract.deployed();
```

Comment:
Oh dear, name reuse here!  Above you use `instance` for what appears to be the class, but here `MyContract` is the class and `instance` is in fact the instance... better make this consistent!